### ISSUE TITLE
Flow doesn't correctly validate Functions when used in an instance variable

### ISSUE THREAD

**Issue by Kadrian** [facebook/flow#4701]
```
// @flow

type PropsType = {
  performAction: () => void,
};

class Comp {
  props: PropsType;

  render() {
    // EXPECT ERROR HERE:
    this.props.performAction.blake.bloke
  }
}
export default Comp;
```

**Test case**
https://flow.org/try/#0PTAEAEDMBsHsHcBQiAuBPADgU1ABQE6wYDOAKpjgLygDeioo2+ks+AtgIIDGKAlrADsAXKAAUASlCUAfKABusXgBMANIgC+AbmRdoAQ2LFQAYVhsMteo0IkRBImQrar+LAKVZ8EywwYhQAKIAGrgBxqSBAEqRAPKRoAASAZEBQlYMKAAWvMQAdBg2eUws7Nx8grkARvoA1lhVcHVW6hqIWAAeGKwooB6QegCu0D2m5tpAA

**Comment by asolove** [comment]
I believe the issue here is that by default, `Function` is a subtype of `Object`, and Flow allows objects to have and use extra, undeclared properties.

Currently, if you want to say that a function object can _only_ be used as a function, and should error if other properties are accessed, you can use this syntax:

```js
type PropsType = {
  performAction: {(): void}
};
```

**Comment by Kadrian** [comment]
Hmm, thanks for that workaround. To me it sounds though that it would be valuable to ask whether `Function` should really be a subtype of `Object`, i.e. in which cases this is actually beneficial.

**Comment by asolove** [comment]
Yup, this same issue and workaround and question of default status actually came up in another issue yesterday. I think we can close this one unless there are any objections.

**Comment by Kadrian** [comment]
Which one?

**Comment by Mathspy** [comment]
This has been fixed since [v0.99.0](https://github.com/facebook/flow/releases/tag/v0.99.0) and it throws the expected error now

@goodmind
